From d0992ca8bf59ecb9d282553c8bbd5e8cc43d8057 Mon Sep 17 00:00:00 2001
From: Neal Liu <neal_liu@aspeedtech.com>
Date: Tue, 22 Jul 2025 10:30:28 +0800
Subject: [PATCH 01/32] arch: arm: ast2600: cpu-info: fix bit shifting for hash
 and rsa

Correct bit shifting for hash and rsa in security info print.

Signed-off-by: Neal Liu <neal_liu@aspeedtech.com>
Change-Id: I62614185c4c40373546b25ee902a5d29d0fc809c
(cherry picked from commit 4ff3a97f45898c9ccde85c81ea042414cdc900be)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 arch/arm/mach-aspeed/ast2600/cpu-info.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-aspeed/ast2600/cpu-info.c b/arch/arm/mach-aspeed/ast2600/cpu-info.c
index b9f20d1f1f..36bd09296a 100644
--- a/arch/arm/mach-aspeed/ast2600/cpu-info.c
+++ b/arch/arm/mach-aspeed/ast2600/cpu-info.c
@@ -96,8 +96,8 @@ void aspeed_print_security_info(void)
 		return;
 	printf("Secure Boot: ");
 	if (qsr & BIT(7)) {
-		hash = (qsr >> 10) & 3;
-		rsa = (qsr >> 12) & 3;
+		hash = (qsr >> 12) & 3;
+		rsa = (qsr >> 10) & 3;
 
 		printf("Mode_2, ");
 
-- 
2.34.1

